core/src/main/java/org/elasticsearch/index/query/IdsQueryParser.java

```diff
@@ -32,6 +33,11 @@
  */
 public class IdsQueryParser implements QueryParser<IdsQueryBuilder> {
 
+    public static final ParseField TYPES_FIELD = new ParseField("type", "types", "_type");
+    // TODO weird: combinations types/type and _type/type were in the original code, but not _type/types/type??
```

Comment:
To me it looks like type/types is for the array version, _type/type for the single key-value version. So I'm not sure if "types" should count as deprecated name.